src/transport.js

```diff
@@ -80,6 +80,9 @@ module.exports = function (swarm) {
       const createListeners = multiaddrs.map((ma) => {
         return (cb) => {
           const listener = transport.createListener(handler)
+
+          listener.once('error', cb)
+
```

Comment:
This event listener needs to unregistered if the event is successful, otherwise a future error (like change of wifi) will make the callback be called a second time with an error